pkg/infra/usagestats/usage_stats_test.go

```diff
@@ -526,3 +528,40 @@ func (aum *alertingUsageMock) QueryUsageStats() (*alerting.UsageStats, error) {
 		},
 	}, nil
 }
+
+func setupSomeDataSourcePlugins(t *testing.T) {
+	plugins.DataSources = make(map[string]*plugins.DataSourcePlugin)
+
+	plugins.DataSources[models.DS_ES] = &plugins.DataSourcePlugin{
+		FrontendPluginBase: plugins.FrontendPluginBase{
+			PluginBase: plugins.PluginBase{
+				Signature: "internal",
+			},
+		},
+	}
+	plugins.DataSources[models.DS_PROMETHEUS] = &plugins.DataSourcePlugin{
+		FrontendPluginBase: plugins.FrontendPluginBase{
+			PluginBase: plugins.PluginBase{
+				Signature: "internal",
+			},
+		},
+	}
+
+	plugins.DataSources[models.DS_GRAPHITE] = &plugins.DataSourcePlugin{
+		FrontendPluginBase: plugins.FrontendPluginBase{
+			PluginBase: plugins.PluginBase{
+				Signature: "internal",
+			},
+		},
+	}
+
+	plugins.DataSources[models.DS_MYSQL] = &plugins.DataSourcePlugin{
+		FrontendPluginBase: plugins.FrontendPluginBase{
+			PluginBase: plugins.PluginBase{
+				Signature: "internal",
+			},
+		},
+	}
+
+	t.Cleanup(func() { plugins.DataSources = nil })
```

Comment:
It's better to move this to the beginning of the function, so you're sure it gets called, otherwise if you're unlucky this function might not get called after you have changed plugins.

Also, I think the following is safer, in case something relies on a default value for `plugins.DataSources`:

```suggestion
origDataSources = plugins.DataSources
t.Cleanup(func() { plugins.DataSources = origDataSources })
```

Reply:
Oh, yes, that's true! My fault! 😅  

For the default value, it's actually `nil`, but I agree it's always better to just capture the previous value and set it again at the end.

Thanks!! 🙏🏻 😇